tests/com/carolinarollergirls/scoreboard/defaults/DefaultClockModelTests.java

```diff
@@ -79,149 +110,327 @@ public void test_defaults() {
 		assertEquals(ID, clock.getProviderId());
 		assertEquals(Clock.class, clock.getProviderClass());
 	}
-	
+
 	@Test
-	public void syncTimeTest() {
+	public void test_reset() {
+		clock.setMaximumNumber(5);
+		clock.setMinimumNumber(2);
+		clock.setNumber(4);
+		clock.setMaximumTime(1200000);
+		clock.setTime(5000);
+		clock.setCountDirectionDown(true);
+		
+		clock.reset();
 		
+		assertTrue(clock.isCountDirectionDown());
+		assertEquals(clock.getMinimumNumber(), clock.getNumber());
+		assertTrue(clock.isTimeAtStart());
 	}
+	
+	@Test
+	public void syncTimeTest() {
+		//add a master clock
+		DefaultClockModel clock2 = new DefaultClockModel(sbModelMock, Clock.ID_TIMEOUT);
+		syncStatus = true;
+		clock.setMaximumTime(10000);
+		clock2.setMaximumTime(10000);
+		clock2.setTime(3400);
+		assertEquals(3400, clock2.getTime());
+		
+		//no syncing is done if the clock is stopped
+		clock.setTime(4200);
+		assertEquals(4200, clock.getTime());
+
+		//when the clocks are started the non-master clock is synced to the master clock
+		clock2.start();
+		clock.start();
+		assertEquals(4400, clock.getTime());
+
+		//changes under 1s are ignored. Even if multiple changes accumulate to more than 1s
+		clock.changeTime(500);
+		clock.changeTime(800);
+		assertEquals(4400, clock.getTime());
+
+		//changes over 1s are rounded down
+		clock.changeTime(1100);
+		assertEquals(5400, clock.getTime());
 
+		//the previous statements also apply to the master clock
+		clock2.changeTime(500);
+		clock2.changeTime(800);
+		assertEquals(3400, clock2.getTime());
+		clock2.changeTime(1000);
+		assertEquals(4400, clock2.getTime());
+
+		//advancing the time affects both clocks even if less than 1s
+		advance(400);
+		assertEquals(4800, clock2.getTime());
+		assertEquals(5800, clock.getTime());
+	}
 	
 	@Test
 	public void test_name() {
+		clock.addScoreBoardListener(new ConditionalScoreBoardListener(clock, Clock.EVENT_NAME, listener));
+		
 		clock.setName("Test Clock");
+		advance(0);
 		
 		assertEquals("Test Clock", clock.getName());
-		//TODO: Test Event system
+		assertEquals(1, collectedEvents.size());
+		ScoreBoardEvent event = collectedEvents.poll();
+		assertEquals("Test Clock", event.getValue());
+		assertEquals(null, event.getPreviousValue());
+	}
+	
+	public void test_count_direction_setter() {
+		assertFalse(clock.isCountDirectionDown());
+		clock.addScoreBoardListener(new ConditionalScoreBoardListener(clock, Clock.EVENT_DIRECTION, listener));
+		
+		clock.setCountDirectionDown(true);
+		advance(0);
```

Comment:
advance(0) seems a bit weird, maybe it'd be clearer to say that you're waiting for the events to propagate?

Reply:
I copied that from you. :)
And after using it 5 times it becomes pretty natural.